graylog2-server/src/main/java/org/graylog2/plugin/Message.java

```diff
@@ -309,7 +317,7 @@ public void addField(final String key, final Object value) {
         } else if(value instanceof String) {
             final String str = ((String) value).trim();
 
-            if(!str.isEmpty()) {
+            if (isRequiredField || !str.isEmpty()) {
```

Comment:
Why would we handle "required" fields different from "normal" fields in this case? Wouldn't this introduce another inconsistency?